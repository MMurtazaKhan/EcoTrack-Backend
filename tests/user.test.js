const request = require("supertest")

const baseUrl = 'https://ecotrack-dev.vercel.app';
const Token = "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpZCI6IjY1ZDk5YmVmMTNlMjYwNjhiMDJhMGIzZSIsImlhdCI6MTcwOTA0MDAwNCwiZXhwIjoxNzExNjMyMDA0fQ.pqa-hj3tF_iDMJVhc8WIgBCVgUOe__aNptQxD9J-GlI";
const UserId = "65f6cf15a0a6a67d2493d722";

describe('User API tests', () => {
  it('should register a new user', async () => {
    const newUser = {
      name: 'Test User',
      email: 'testuser@example.com',
      password: 'password123',
      profilePic: 'http://example.com/profile-pic.jpg'
    };

    const response = await request(baseUrl)
      .post('/api/users/register')
      .send(newUser);

    expect(response.status).toBe(201);
    expect(response.body).toMatchSnapshot();
  });

  it('should authenticate a user', async () => {
    const user = {
      email: 'testuser@example.com',
      password: 'password123'
    };

    const response = await request(baseUrl)
      .post('/api/users/login')
      .send(user);

    expect(response.status).toBe(200);
    expect(response.body).toMatchSnapshot();
  });

  it('should get all users', async () => {
    const response = await request(baseUrl).get('/api/users');

    expect(response.status).toBe(200);
    expect(response.body).toMatchSnapshot();
  });

  it('should get a specific user by ID', async () => {
    const userId = UserId;

    const response = await request(baseUrl).get(`/api/users/${userId}`);

    expect(response.status).toBe(200);
    expect(response.body).toMatchSnapshot();
  });

  it('should edit a user profile', async () => {
    const token = Token;
    const updatedData = {
      name: 'Updated Test User',
      email: 'updatedtestuser@example.com'
    };

    const response = await request(baseUrl)
      .put('/api/users/edit')
      .set('Authorization', `Bearer ${Token}`)
      .send(updatedData);

    expect(response.status).toBe(200);
    expect(response.body).toMatchSnapshot();
  });

  it('should delete all users', async () => {
    const response = await request(baseUrl).delete('/api/users');

    expect(response.status).toBe(200);
    expect(response.body).toMatchSnapshot();
  });

});
