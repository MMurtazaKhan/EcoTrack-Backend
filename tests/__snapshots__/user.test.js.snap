// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User API tests should authenticate a user 1`] = `
{
  "__v": 0,
  "_id": "667e94a8442a4c546e87b17c",
  "companyName": "false",
  "createdAt": "2024-06-28T10:47:04.113Z",
  "creditLimit": 0,
  "email": "testuser@example.com",
  "name": "Test User",
  "profilePic": "http://example.com/profile-pic.jpg",
  "rewardHistory": [],
  "role": "user",
  "token": "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpZCI6IjY2N2U5NGE4NDQyYTRjNTQ2ZTg3YjE3YyIsImlhdCI6MTcxOTU3MTYyNCwiZXhwIjoxNzIyMTYzNjI0fQ.2mAH2SxZcR2foCBCgttw0enpPHlcxXo1p5qOs7_TOIo",
  "updatedAt": "2024-06-28T10:47:04.113Z",
  "virtualCoins": 0,
}
`;

exports[`User API tests should delete all users 1`] = `
{
  "message": "All users deleted successfully",
}
`;

exports[`User API tests should get all users 1`] = `
[
  {
    "__v": 0,
    "_id": "667e94a8442a4c546e87b17c",
    "companyName": "false",
    "createdAt": "2024-06-28T10:47:04.113Z",
    "creditLimit": 0,
    "email": "testuser@example.com",
    "name": "Test User",
    "password": "$2a$10$Cu.hKFKCvVxMwdg6iDWUoerNmcZLCF1ZfbVfoAWXc3ZM6wa7KqkgK",
    "profilePic": "http://example.com/profile-pic.jpg",
    "rewardHistory": [],
    "role": "user",
    "updatedAt": "2024-06-28T10:47:04.113Z",
    "virtualCoins": 0,
  },
]
`;

exports[`User API tests should register a new user 1`] = `
{
  "__v": 0,
  "_id": "667e94a8442a4c546e87b17c",
  "companyName": "false",
  "createdAt": "2024-06-28T10:47:04.113Z",
  "creditLimit": 0,
  "email": "testuser@example.com",
  "name": "Test User",
  "profilePic": "http://example.com/profile-pic.jpg",
  "rewardHistory": [],
  "role": "user",
  "token": "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpZCI6IjY2N2U5NGE4NDQyYTRjNTQ2ZTg3YjE3YyIsImlhdCI6MTcxOTU3MTYyNCwiZXhwIjoxNzIyMTYzNjI0fQ.2mAH2SxZcR2foCBCgttw0enpPHlcxXo1p5qOs7_TOIo",
  "updatedAt": "2024-06-28T10:47:04.113Z",
  "virtualCoins": 0,
}
`;
